### PR TITLE
Examination page speciality

### PR DESCRIPTION
**Pull request by MarkC33**
Added dropdown / select for examination speciality. Also includes fixes for broken unit tests.

### CODE REVIEW

File: Mep.Ui/src/app/services/simple-list/simple-list.service.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import { Observable } from 'rxjs';
+import { HttpClient } from '@angular/common/http';
+import { environment } from '../../../environments/environment';
+import { map } from 'rxjs/operators';
+import { SimpleList } from 'src/app/interfaces/simple-list';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class SimpleListService {
```
Comment (neildavies11) [inline Mep.Ui/src/app/services/simple-list/simple-list.service.ts]: I think we should call this and the interface name-id-list as its more explicit than simple-list and explains exactly what's in it. I think there may be a tendancy in the future to increase the number of properties of the "simple" list if it's named as such.

File: Mep.Ui/src/app/components/examination/examination-create/examination-create.component.ts [hunk 2]
```diff
@@ -77,6 +81,12 @@ export class ExaminationCreateComponent implements OnInit {
       })
     );
 
+    // get the list of specialities for the dropdown
+    this.simpleListService.GetListData('speciality')
+      .subscribe(specialities => {
+        this.specialities = specialities;
+      });
```
[the remark this answers is not in the document]
Comment (MarkC33) [reply Mep.Ui/src/app/components/examination/examination-create/examination-create.component.ts]: toasts rewritten and added to the speciality data subscription

File: Mep.Ui/src/app/services/simple-list/simple-list.service.ts (hunk 1 above)
Comment (MarkC33) [reply Mep.Ui/src/app/services/simple-list/simple-list.service.ts]: interface name updated

File: Mep.Ui/src/app/components/toasts/toasts.component.html [hunk 3]
```diff
@@ -1,12 +1,35 @@
 <ngb-toast
   *ngFor="let toast of toastService.toasts"
   [class]="toast.classname"
-  [autohide]="true"
+  [autohide]="toast.hide || true"
```
[the remark this answers is not in the document]
Comment (MarkC33) [reply Mep.Ui/src/app/components/toasts/toasts.component.html]: changed

File: Mep.Ui/src/app/interfaces/toast-options.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+export interface ToastOptions {
+  classname?: string;
+  delay?: number;
+  hide?: boolean;
```
[the remark this answers is not in the document]
Comment (MarkC33) [reply Mep.Ui/src/app/interfaces/toast-options.ts]: changed

**Pull request merged**